core/geometry/src/polyface/PolyfaceQuery.ts

```diff
@@ -273,6 +274,74 @@ export class PolyfaceQuery {
     return MomentData.inertiaProductsToPrincipalAxes(origin, inertiaProducts);
   }
 
+  /**
+   * Test for convex volume by dihedral angle tests on all edges.
+   *
+   * @param source mesh to examine
+   * @returns true if the mesh is closed and has all dihedral angles (angle across edge) positive
+   */
+  public static isConvexByDihedralAngleCount(source: Polyface): boolean {
+    return this.dihedralAngleSummary(source) > 0;
+  }
+
+  /**
+  * * Return a number summarizing the dihedral angles in the mesh.
+  *   * Return 1 if all angles are positive or planar.  The mesh is probably convex with outward normals.
+  *   * Return -1 if all angles are negative or planar.  The mesh is probably convex with inward normals.
+  *   * Return 0 if
+  *     * angles area mixed
+  *     * any edge has other than 1 incident facet or more than 2 incident facets.
+  *     * (but null edges are permitted -- These occur naturally at edges of quads at north or south pole)
+  */
+  public static dihedralAngleSummary(source: Polyface): number {
+    {
+      const edges = new IndexedEdgeMatcher();
+      const visitor = source.createVisitor(1);
+      visitor.reset();
+      const centroidNormal: Ray3d[] = [];
+      let normalCounter = 0;
+      while (visitor.moveToNextFacet()) {
+        const numEdges = visitor.pointCount - 1;
+        const normal = PolygonOps.centroidAreaNormal(visitor.point);
+        if (normal === undefined)
+          return 0;
+        centroidNormal.push(normal);
+        for (let i = 0; i < numEdges; i++) {
+          edges.addEdge(visitor.clientPointIndex(i), visitor.clientPointIndex(i + 1), normalCounter);
+        }
+        normalCounter++;
+      }
+      const badClusters: SortableEdgeCluster[] = [];
+      const manifoldClusters: SortableEdgeCluster[] = [];
+
+      edges.sortAndCollectClusters(manifoldClusters, badClusters, undefined, badClusters);
+      if (badClusters.length > 0)
+        return 0;
```

Comment:
Would be useful if this function worked on non-closed meshes too. I think all you'd need is to set 2nd arg here to undefined (to ignore singleton edges). Maybe make this optional: add a "mustBeClosed" arg to `dihedralAngleSummary` that determines whether the 2nd arg is undefined, and for which `isConvexByDihedralAngleCount` could pass true.


This test be extended to test non-closed meshes by ignoring singleton edges too.

Reply:
See ignoreBoundaries option added in later commits.